src/libs/API.js

```diff
@@ -346,148 +196,175 @@ function getAuthToken() {
 
 /**
  * @param {object} parameters
+ * @param {string} [parameters.useExpensifyLogin]
+ * @param {string} parameters.partnerName
+ * @param {string} parameters.partnerPassword
  * @param {string} parameters.partnerUserID
  * @param {string} parameters.partnerUserSecret
- * @param {string} parameters.twoFactorAuthCode
- * @param {string} [parameters.exitTo]
+ * @param {string} [parameters.twoFactorAuthCode]
  * @returns {Promise}
  */
-function authenticate(parameters) {
-    Onyx.merge(ONYXKEYS.SESSION, {loading: true, error: ''});
-
-    // We treat Authenticate in a special way because unlike other commands, this one can't fail
-    // with 407 authToken expired. When other api commands fail with this error we call Authenticate
-    // to get a new authToken and then fire the original api command again
-    return HttpUtils.xhr('Authenticate', {
-        // When authenticating for the first time, we pass useExpensifyLogin as true so we check for credentials for
-        // the expensify partnerID to let users authenticate with their expensify user and password.
-        useExpensifyLogin: true,
-        partnerName: CONFIG.EXPENSIFY.PARTNER_NAME,
-        partnerPassword: CONFIG.EXPENSIFY.PARTNER_PASSWORD,
+function Authenticate(parameters) {
+    const commandName = 'Authenticate';
+
+    requireParameters([
+        'partnerName',
+        'partnerPassword',
+        'partnerUserID',
+        'partnerUserSecret',
+    ], parameters, commandName);
```

Comment:
I think we should handle the thrown error by `requireParameters` here to redirect a user to login page. Issue https://github.com/Expensify/App/issues/58384